### DISCUSSION TITLE
ScriptReference/PlayerSettings.GetPreloadedAssets

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#615] [Notes]
# ScriptReference/PlayerSettings.GetPreloadedAssets

https://docs.unity3d.com/ScriptReference/PlayerSettings.GetPreloadedAssets.html

**Comment by bottinogames** [comment]
The example code included has one minor flaw, notably, preloaded assets are only preloaded in builds, and ***not*** when in the editor (as of 2022.3). 
Assets will ***only*** be loaded in editor the first time they are referenced, meaning that until a scriptable object is referenced in a scene or viewed in the inspector, `OnEnable` will not be called. 
Thus, in the example code, `configInstance` will be null if the editor is restarted, until the ScriptableObject is viewed with an inspector.

Calling `PlayerSettings.GetPreloadedAssets()` in the editor will load all preload assets which have not previously been loaded.

The provided example code can be fixed by adding this small bit of code to ensure assets are loaded when the editor is restarted:
```cs
#if UNITY_EDITOR
[InitializeOnLoadMethod]
public static void EditorInitializeOnLoad()
{
    // Force unity to load preloaded assets on editor start
    _ = PlayerSettings.GetPreloadedAssets();
}
#endif
```
